hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -91,412 +90,710 @@ public class SingleTransactionRecordBuilderImpl
+    public SingleTransactionRecordBuilderImpl alias(@NonNull final Bytes alias) {
+        requireNonNull(alias, "alias must not be null");
+        this.transactionRecordBuilder.alias(alias);
         return this;
     }
 
-    public SingleTransactionRecordBuilderImpl ethereumHash(Bytes ethereumHash) {
-        this.ethereumHash = ethereumHash;
+    /**
+     * Sets the ethereum hash.
+     *
+     * @param ethereumHash the ethereum hash
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl ethereumHash(@NonNull final Bytes ethereumHash) {
+        requireNonNull(ethereumHash, "ethereumHash must not be null");
+        this.transactionRecordBuilder.ethereumHash(ethereumHash);
         return this;
     }
 
-    public SingleTransactionRecordBuilderImpl paidStakingRewards(List<AccountAmount> paidStakingRewards) {
+    /**
+     * Sets the paidStakingRewards.
+     *
+     * @param paidStakingRewards the paidStakingRewards
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl paidStakingRewards(
+            @NonNull final List<AccountAmount> paidStakingRewards) {
+        requireNonNull(paidStakingRewards, "paidStakingRewards must not be null");
         this.paidStakingRewards = paidStakingRewards;
         return this;
     }
 
+    /**
+     * Adds a paidStakingReward.
+     *
+     * @param paidStakingReward the paidStakingReward
+     * @return the builder
+     */
     @NonNull
-    public SingleTransactionRecordBuilderImpl entropyNumber(final int num) {
-        this.entropy = new OneOf<>(EntropyOneOfType.PRNG_NUMBER, num);
+    public SingleTransactionRecordBuilderImpl addPaidStakingReward(@NonNull final AccountAmount paidStakingReward) {
+        requireNonNull(paidStakingReward, "paidStakingReward must not be null");
+        this.paidStakingRewards.add(paidStakingReward);
         return this;
     }
 
+    /**
+     * Sets the entropy to a given number.
+     *
+     * @param num number to use for entropy
+     * @return the builder
+     */
     @NonNull
-    public SingleTransactionRecordBuilderImpl entropyBytes(@NonNull final Bytes prngBytes) {
-        requireNonNull(prngBytes, "The argument 'entropyBytes' must not be null");
-        this.entropy = new OneOf<>(EntropyOneOfType.PRNG_BYTES, prngBytes);
+    public SingleTransactionRecordBuilderImpl entropyNumber(final int num) {
+        this.transactionRecordBuilder.prngNumber(num);
         return this;
     }
 
     /**
-     * @deprecated this method is only used temporarily during the migration
+     * Sets the entropy to given bytes.
+     *
+     * @param prngBytes bytes to use for entropy
+     * @return the builder
      */
-    @Deprecated(forRemoval = true)
-    @Nullable
-    public OneOf<TransactionRecord.EntropyOneOfType> entropy() {
-        return entropy;
+    @NonNull
+    public SingleTransactionRecordBuilderImpl entropyBytes(@NonNull final Bytes prngBytes) {
```

Comment:
As with many methods, this is actually an Override from one of the interfaces.
Could we add all of the missing `@Overrides` annotations (and perhaps make sure they are all in consistent order with Override before NonNull if both are present)?

Reply:
Done.

Reply:
Btw out of curiosity, why the order should be Override before NonNull?

Reply:
Just to be consistent with the wider codebase; it's the most common pre-existing order.